odoo/addons/base/models/ir_ui_view.py

```diff
@@ -402,11 +425,6 @@ def create(self, values):
 
     @api.multi
     def write(self, vals):
-        # If view is modified we remove the arch_fs information thus activating the arch_db
-        # version. An `init` of the view will restore the arch_fs for the --dev mode
-        if ('arch' in vals or 'arch_base' in vals) and 'install_mode_data' not in self._context:
-            vals['arch_fs'] = False
```

Comment:
fair note: I do not exactly know what the removal of this piece of code entails, but it is necessary to keep the arch_fs intact in order to restore the original view

cc @odony

Reply:
after doing some digging, I found out why this was done: the `xml` --dev option never properly worked, everything is explained (and fixed) in the first commit message.